File: ERP-Inventario/Clases/CSql.php
<?php
// Conexion a Base de Datos
class CSql {
   public $pcError, $pnNumRow;
   protected $h;
 
   public function __construct() {
      $this->pcError = null;
      $this->pnNumRow = 0;
   }

   public function omDisconnect() {
      $this->omExec("COMMIT;");
      pg_close($this->h); 
   }
 
   public function omConnect($lnFlag = 0) {
      if ($lnFlag == 1) {
         $lcConStr = "host=localhost dbname=UCSMListener port=5432 user=postgres password=postgres";
      } else if ($lnFlag == 2) {
         $lcConStr = "host=localhost dbname=UCSMINS port=5432 user=postgres password=postgres";
      } else if ($lnFlag == 3) {
         $lcConStr = "host=localhost dbname=UCSMASBANC port=5432 user=postgres password=postgres";
      } else if ($lnFlag == 4) {
         $lcConStr = "host=localhost dbname=UCSMFactElec port=5432 user=postgres password=postgres";
      } else if ($lnFlag == 5) {
         $lcConStr = "host=localhost dbname= port=5432 user=postgres password=postgres";
      } else if ($lnFlag == 7) {
         $lcConStr = "host=localhost dbname=UCSMERP_DW port=5432 user=postgres password=postgres";
      } else if ($lnFlag == 8) {
         $lcConStr = "host=localhost dbname=UcsmFactElec port=5432 user=postgres password=postgres";
      } else if ($lnFlag == 10) {
         $lcConStr = "host=localhost dbname=ISPSMERP port=5432 user=postgres password=postgres";
      } else if ($lnFlag == 11) {
         $lcConStr = "host=localhost dbname=ISPSMFACT port=5432 user=postgres password=postgres";
      } else if ($lnFlag == 12) {
         $lcConStr = "host=localhost dbname=UCSMACR port=5432 user=postgres password=postgres";
      } else if ($lnFlag == 13) {
         $lcConStr = "host=localhost dbname=UCSMAFJ port=5432 user=postgres password=postgres";
      } else if ($lnFlag == 14) {
         $lcConStr = "host=localhost dbname=UCSMDEEP port=5432 user=postgres password=postgres";
      } else {
         $lcConStr = "host=localhost dbname=UCSMERP port=5432 user=postgres password=postgres";
      }
      try {
         @$this->h = pg_connect($lcConStr);
      } catch (Exception $ex) {
         $this->pcError = "No se pudo conectar a la base de datos";
         return false;
      }
      $this->omExec("BEGIN;");
      return true;
   }

   public function omExec($p_cSql) {
      $lcSql = substr(strtoupper(trim($p_cSql)), 0, 6);
      if ($lcSql === "SELECT") {
         $this->pnNumRow = 0;
         $RS = pg_query($this->h, $p_cSql);
         if (!($RS)) {
            $this->pcError = "Error al ejecutar comando SQL";
            return false;
         }
         $this->pnNumRow = pg_num_rows($RS);
         return $RS;
      } else {
         @$RS = pg_query($this->h, $p_cSql);
         if (pg_affected_rows($RS) === 0)                                                 
            if (!($RS)) {
               $this->pcError = "La operacion no afecto a ninguna fila";
               return false;
            }
         return true;
      }
   }

   public function fetch($RS) {
      return pg_fetch_row($RS);     
   }
   
   public function rollback() {
      $this->omExec("ROLLBACK;");
   }
}

?>

File: ERP-Inventario/ERPWS/Clases/CSql.py
import hashlib
import psycopg2
# import pymssql
import json

class CSql():
    def __init__(self):
        self.h       = None
        self.plOk    = True
        self.pcError = None 

    def omConnect(self, p_nDB = None):
        self.plOk = True
        lcConnect = "host=localhost dbname=UCSMERP port=5432 user=postgres password=postgres"
        if p_nDB == 1:
           lcConnect = "host=10.0.7.160 dbname=UCSMListener user=postgres password=12Listen34 port=5432"
        elif p_nDB == 2:
           lcConnect = "host=localhost dbname=UCSMINS port=5432 user=ucsmerpweb password=7d88531d42f806ec28a446c3fe0448ee"
        elif p_nDB == 3:
           lcConnect = "host=localhost dbname=UCSMASBANC user=postgres password=postgres port=5432"
        elif p_nDB == 4:
           lcConnect = "host=10.0.7.160 dbname=UCSMFactElec user=postgres password=12Listen34 port=5432"  
        elif p_nDB == 6:
           lcConnect = "host=10.0.7.170 dbname=UCSMERP_DW port=5432 user=ucsmerpweb password=12UC\$M3RP34"
        elif p_nDB == 7:
           lcConnect = "host=10.10.110.13 dbname=UCSMFactElec port=5432 user=postgres password=PostgresAliviari159"
        elif p_nDB == 8:
           lcConnect = "host=localhost dbname=UCSMERP port=5432 user=postgres password=postgres"
        try:
           self.h = psycopg2.connect(lcConnect) 
        except psycopg2.DatabaseError:
           self.plOk = False
           self.pcError = 'ERROR AL CONECTAR CON LA BASE DE DATOS'
        return self.plOk

    def omExecRS(self, p_cSql):
        #print p_cSql
        self.plOk = True
        lcCursor = self.h.cursor()
        try:
           lcCursor.execute(p_cSql)
           RS = lcCursor.fetchall()
        except psycopg2.DatabaseError as e:
           self.plOk = False
           #print e.message
           self.pcError = 'ERROR AL EJECUTAR COMANDO SELECT'
           RS = None
        return RS

    def omExec(self, p_cSql):
        #print p_cSql
        self.plOk = True
        lcCursor = self.h.cursor()
        try:
           lcCursor.execute(p_cSql)
        except psycopg2.DatabaseError as e:
           self.plOk = False
     #      print e.message
           self.pcError = 'ERROR AL ACTUALIZAR LA BASE DE DATOS'
        return self.plOk

    def omDisconnect(self):
        self.h.close()

    def omCommit(self):
        self.h.commit()

class CSqlServer():
   def __init__(self):
       self.h       = None
       self.plOk    = True
       self.pcError = None

   def omConnect(self, p_nDB = 0):
       self.plOk = True
       try:
          if p_nDB == 1:
             # DB Fotografias
             self.h = pymssql.connect("10.0.2.78\PICDB", "userAppUCSM", "4pp$UcSm", "UCSM_PIC")
          else:
             # DB Matriculas
             self.h = pymssql.connect("10.0.2.61:1433\SVRDB01", "userAppUCSM", "4pp$UcSm", "UCSM")
       except:
          self.plOk = False
          self.pcError = 'ERROR AL CONECTAR CON SQL-SERVER'
       return self.plOk

   def omExecRS(self, p_cSql):
       #print p_cSql
       self.plOk = True
       lcCursor = self.h.cursor()
       try:
          lcCursor.execute(p_cSql)
          RS = lcCursor.fetchall()
       except pymssql.DatabaseError as e:
      #    print e.message
          self.plOk = False
          self.pcError = 'ERROR AL EJECUTAR COMANDO SQL'
          RS = None
       return RS

   def omExec(self, p_cSql):
       self.plOk = True
       lcCursor = self.h.cursor()
       try:
          lcCursor.execute(p_cSql)
       except pymssql.DatabaseError as e:
          self.plOk = False
          self.pcError = 'ERROR AL ACTUALIZAR BASE DE DATOS'
       #   print e.message
       return self.plOk

   def omDisconnect(self):
       self.h.close()

   def omCommit(self):
       self.h.commit()
